examples/lighting-app/nrfconnect/main/main.cpp

```diff
@@ -21,16 +21,51 @@
 #include <platform/CHIPDeviceLayer.h>
 #include <support/CHIPMem.h>
 
+#include <kernel.h>
+
+#ifndef CONFIG_CHIP_PW_RPC
 #include <logging/log.h>
+LOG_MODULE_DECLARE(app);
+#else
+#include "pw_log/log.h"
+#include "pw_sys_io/sys_io.h"
+#include "pw_sys_io_nrfconnect/init.h"
 
-LOG_MODULE_REGISTER(app);
+#define LOG_INF(message, ...) PW_LOG_INFO(message, __VA_ARGS__)
+#define LOG_ERR(message, ...) PW_LOG_ERROR(message, __VA_ARGS__)
+#endif
 
 using namespace ::chip;
 using namespace ::chip::Inet;
 using namespace ::chip::DeviceLayer;
 
+#ifdef CONFIG_CHIP_PW_RPC
+#pragma message "Building with RPC support"
+
+namespace chip {
+namespace rpc {
+extern void RunRpcService(void *, void *, void *);
+}
```

Comment:
// namespace rpc

would it make sense to create a header for this? having extern inside .cpp file seems a tad  ugly

Reply:
created Rpc.h